network/README.md

```diff
@@ -51,80 +51,85 @@ partial membership views, sophisticated failure detectors, or network overlays.
 
 ### System Architecture
 
-    +--------------------+----------------------+
-    |      Consensus     |        Mempool       |
-    +--------------------+----------------------+
-    |               Peer Manager                |
-    +-------------------------------------------+
-    |              NetworkProvider(s)           |
-    +--------------------+----------------------+
-    |        RPC(s)      |      DirectSend(s)   |
-    +--------------------+----------------------+
-    |                  Peer(s)                  |
-    +-------------------------------------------+
+```
++-----------+---------+------------+--------+
+| Consensus | Mempool | State Sync | Health |  Application Modules
++-----------+---------+------------+--------+
+      ^          ^          ^           ^
+      |          |          |           |      Network Interface
+      v          v          v           v
++----------------+--------------------------+
+| Peer Manager  <->   Connectivity Manager  |  Network Module
++----------------+--------------------------+
+|    Peer(s)     |
++----------------+
```

Comment:
Might be more appropriate to put the connectivity manager directly to the side, everything really kind of talks to Peer Manager except for discovery

```
+-----------+---------+------------+--------+
| Consensus | Mempool | State Sync | Health |  Application Modules
+-----------+---------+------------+--------+
      ^          ^          ^           ^
      |          |          |           |      Network Interface
      v          v          v           v
+------------------------------------------+   +----------------------+
|              Peer Manager                |<->| Connectivity Manager |  Network Module
+------------------------------------------+   +----------------------+
|    Peer(s)     |
+----------------+
```